[BUG] non-utc integration tests failing in json_test.py

We are seeing some integration test failures when the timezone isn't UTC:

```
[2024-09-18T14:17:46.633Z] FAILED ../../src/main/python/json_test.py::test_basic_from_json[yyyy-MM-dd-false-false-false-StructType(List(StructField(number,DateType,true)))-floats_edge_cases.json][DATAGEN_SEED=1726661844, TZ=Canada/Newfoundland, APPROXIMATE_FLOAT, ALLOW_NON_GPU(FileSourceScanExec,BatchScanExec,FileSourceScanExec)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
[2024-09-18T14:17:46.633Z] FAILED ../../src/main/python/json_test.py::test_basic_from_json[yyyy-MM-dd-false-false-false-StructType(List(StructField(number,DateType,true)))-decimals.json][DATAGEN_SEED=1726661844, TZ=Canada/Newfoundland, INJECT_OOM, APPROXIMATE_FLOAT, ALLOW_NON_GPU(FileSourceScanExec,BatchScanExec,FileSourceScanExec)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
[2024-09-18T14:17:46.633Z] FAILED ../../src/main/python/json_test.py::test_basic_from_json[yyyy-MM-dd-false-false-false-StructType(List(StructField(number,DateType,true)))-dates.json][DATAGEN_SEED=1726661844, TZ=Canada/Newfoundland, APPROXIMATE_FLOAT, ALLOW_NON_GPU(FileSourceScanExec,BatchScanExec,FileSourceScanExec)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
```

```
[2024-09-18T12:54:33.809Z] !Exec <ProjectExec> cannot run on GPU because not all expressions can be replaced
[2024-09-18T12:54:33.809Z]   @Expression <Alias> value#105916 AS json#105918 could run on GPU
[2024-09-18T12:54:33.809Z]     @Expression <AttributeReference> value#105916 could run on GPU
[2024-09-18T12:54:33.809Z]   @Expression <Alias> from_json(StructField(number,BooleanType,true), (allowNumericLeadingZeros,true), (allowNonNumericNumbers,true), value#105916, Some(Canada/Newfoundland)) AS from_json(json)#105920 could run on GPU
[2024-09-18T12:54:33.809Z]     !Expression <JsonToStructs> from_json(StructField(number,BooleanType,true), (allowNumericLeadingZeros,true), (allowNonNumericNumbers,true), value#105916, Some(Canada/Newfoundland)) cannot run on GPU because class org.apache.spark.sql.catalyst.expressions.JsonToStructs is not supported with timezone settings: (JVM: Canada/Newfoundland, session: Canada/Newfoundland). Set both of the timezones to UTC to enable class org.apache.spark.sql.catalyst.expressions.JsonToStructs support
[2024-09-18T12:54:33.809Z]       @Expression <AttributeReference> value#105916 could run on GPU
[2024-09-18T12:54:33.809Z]   !Exec <FileSourceScanExec> cannot run on GPU because unsupported file format: org.apache.spark.sql.execution.datasources.text.TextFileFormat
```

--- a/integration_tests/src/main/python/json_matrix_test.py
+++ b/integration_tests/src/main/python/json_matrix_test.py
@@ -17,9 +17,6 @@ import pytest
 
 from asserts import *
 from data_gen import *
-from conftest import is_not_utc
-from datetime import timezone
-from conftest import is_databricks_runtime
 from marks import approximate_float, allow_non_gpu, ignore_order, datagen_overrides
 from spark_session import *
 

--- a/integration_tests/src/main/python/json_test.py
+++ b/integration_tests/src/main/python/json_test.py
@@ -25,6 +25,12 @@ from spark_session import *
 
 TEXT_INPUT_EXEC='FileSourceScanExec'
 
+# allow non gpu when time zone is non-UTC because of https://github.com/NVIDIA/spark-rapids/issues/9653'
+non_utc_file_source_scan_allow = ['FileSourceScanExec'] if is_not_utc() else []
+
+non_utc_project_allow = ['ProjectExec'] if is_not_utc() else []
+
+
 json_supported_gens = [
     # Spark does not escape '\r' or '\n' even though it uses it to mark end of record
     # This would require multiLine reads to work correctly, so we avoid these chars
@@ -378,7 +384,7 @@ def test_basic_json_read(std_input_path, filename, schema, read_func, allow_non_
     'false'
 ])
 @pytest.mark.parametrize('ansi_enabled', ["true", "false"])
-@allow_non_gpu(TEXT_INPUT_EXEC, *not_utc_allow_for_test_json_scan)
+@allow_non_gpu(TEXT_INPUT_EXEC, *non_utc_project_allow)
 @pytest.mark.parametrize('date_format', [None, 'yyyy-MM-dd'])
 def test_basic_from_json(std_input_path, filename, schema, allow_non_numeric_numbers, \
         allow_numeric_leading_zeros, ansi_enabled, date_format):
@@ -581,11 +587,6 @@ def test_json_read_invalid_dates(std_input_path, filename, schema, read_func, an
             conf=updated_conf)
     else:
         assert_gpu_and_cpu_are_equal_collect(f, conf=updated_conf)
-
-# allow non gpu when time zone is non-UTC because of https://github.com/NVIDIA/spark-rapids/issues/9653'
-non_utc_file_source_scan_allow = ['FileSourceScanExec'] if is_not_utc() else []
-
-non_utc_project_allow = ['ProjectExec'] if is_not_utc() else []
 
 @approximate_float
 @pytest.mark.parametrize('filename', [
@@ -817,9 +818,6 @@ def test_from_json_struct_date_fallback_non_default_format(date_gen, date_format
             .select(f.col('a'), f.from_json('a', 'struct<a:date>', options)),
         'ProjectExec',
         conf=conf)
-
-# allow non gpu when time zone is non-UTC because of https://github.com/NVIDIA/spark-rapids/issues/9653'
-non_utc_project_allow = ['ProjectExec'] if is_not_utc() else []
 
 @pytest.mark.parametrize('timestamp_gen', [
     # "yyyy-MM-dd'T'HH:mm:ss[.SSS][XXX]"

